internal/events/events_test.go

```diff
@@ -0,0 +1,45 @@
+package events
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func TestEventLoop_SwapBatches(t *testing.T) {
+	eventLoop := NewEventLoop(nil, zap.New(), nil, nil)
+
+	eventLoop.currentBatch = []interface{}{
```

Comment:
This assignment left me a little puzzled in connection to the one below. I think both assignment should be `EventBatch{}`.

Reply:
Fixed